subsys/net/l2/openthread/openthread.c

```diff
@@ -428,26 +438,30 @@ static int openthread_init(struct net_if *iface)
 		platformShellInit(ot_context->instance);
 	}
 
-	if (IS_ENABLED(CONFIG_OPENTHREAD_NCP)) {
+	if (IS_ENABLED(CONFIG_OPENTHREAD_COPROCESSOR)) {
 		otNcpInit(ot_context->instance);
 	}
 
-	otIp6SetEnabled(ot_context->instance, true);
+	if (!IS_ENABLED(CONFIG_OPENTHREAD_COPROCESSOR_RCP)) {
```

Comment:
I don't think the configuration should differ for NCP/RCP, I'd just make it
```
	if (IS_ENABLED(CONFIG_OPENTHREAD_COPROCESSOR)) {
		otNcpInit(ot_context->instance);
	} else {
		...
	}
```
The only difference is that right now for NCP `otIp6SetEnabled` is called, but it doesn't seem to be needed (it's not called in the sample from the OT upstream repository, it's the host responsibility to configure NCP/RCP).

Reply:
Changed.